test/t/unit/test_unit_xfunc.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+
+from conftest import assert_bash_exec, bash_env_saved
+
+
+@pytest.mark.bashcomp(cmd=None)
+class TestUnitXfunc:
+    def test_1(self, bash):
+        with bash_env_saved(bash) as bash_env:
+            bash_env.write_variable(
+                "BASH_COMPLETION_USER_DIR", "$PWD/_comp_xfunc", quote=False
+            )
+
+            # first invocation (completion/xfunc-test1 is sourced)
+            output = assert_bash_exec(
+                bash,
+                "_comp_xfunc xfunc-test1 utility1 'a b' cde fgh",
+                want_output=True,
+            )
+            assert output.strip() == "util1[<a b><cde><fgh>]"
+
+            # first invocation (completion/xfunc-test1 is not sourced)
+            output = assert_bash_exec(
+                bash,
+                "_comp_xfunc xfunc-test1 utility1 'a b' cde fgh",
+                want_output=True,
+            )
+            assert output.strip() == "util1[<a b><cde><fgh>]"
+
+    def test_2(self, bash):
+        with bash_env_saved(bash) as bash_env:
+            bash_env.write_variable(
+                "BASH_COMPLETION_USER_DIR", "$PWD/_comp_xfunc", quote=False
+            )
+
```

Comment:
Similarly here, perhaps assert that `_comp_xfunc_non_standard_name` is not yet defined?

Reply:
Yes! I will process it in the other [(conversation)](https://github.com/scop/bash-completion/pull/734/files#r848820292),

Reply:
573e0db0